python/tests/plugins/test_gen_design_doc_impl.py

```diff
@@ -1,39 +1,43 @@
 import os
-from tempfile import TemporaryDirectory
 from unittest import TestCase
 
 from aac.plugins.gen_design_doc.gen_design_doc_impl import gen_design_doc
 
 from tests.helpers.assertion import assert_plugin_success
+from tests.helpers.io import temporary_test_file
 
 
 class TestGenerateDesignDocumentPlugin(TestCase):
     def test_can_generate_design_doc_with_models(self):
-        with TemporaryDirectory() as temp_dir:
-            test_model_file_name = f"{temp_dir}/test_model.yaml"
-            test_design_doc_file_name = None
-            with open(test_model_file_name, "w") as arch_file:
-                arch_file.write(TEST_MODEL)
-
-            result = gen_design_doc(test_model_file_name, temp_dir)
-            assert_plugin_success(result)
-
+        with temporary_test_file(TEST_MODEL) as test_model: 
+            temp_dir = os.path.dirname(test_model.name)
+            result = gen_design_doc(test_model.name, temp_dir)
+            assert_plugin_success(result) 
             files = os.listdir(temp_dir)
             self.assertEqual(len(files), 2)
-
-            test_design_doc_file_name, *_ = [
-                f for f in files if f != os.path.basename(test_model_file_name)
-            ]
-            with open(f"{temp_dir}/{test_design_doc_file_name}", "r") as design_doc:
-                markdown = design_doc.read()
-                self.assert_headings(markdown)
+            test_design_doc_file_name, *_ = [f for f in files if f != os.path.basename(test_model.name)]
+            with open(os.path.join(temp_dir, test_design_doc_file_name)) as markdown_file:
+                markdown = markdown_file.read()
+                self.assert_headings(markdown, os.path.basename(test_model.name))
                 self.assert_schema(markdown)
                 self.assert_model(markdown)
                 self.assert_use_case(markdown)
+   
+    def test_can_handle_names_with_dots(self):
+        with temporary_test_file(TEST_MODEL_2) as test_yaml:
+            temp_dir = os.path.dirname(test_yaml.name)
+
+            result = gen_design_doc(test_yaml.name, temp_dir)
+            assert_plugin_success(result)
+            files = os.listdir(temp_dir)
+            self.assertEqual(len(files), 2)
+            test_design_doc_file_name, *_ = [f for f in files if f != os.path.basename(test_yaml.name)]
+            with open(os.path.join(temp_dir, test_design_doc_file_name)) as markdown_file:
+                self.assertIn("SubSchema.Schema1 data", markdown_file.read())
```

Comment:
Since we changed how we handle `required_fields` it would be good to make sure `"required"` doesn't show up in the markdown file contents (since `TEST_MODEL_2` doesn't have any required fields).

```suggestion
                markdown = markdown_file.read()
                self.assertIn("SubSchema.Schema1 data", markdown)
                self.assertNotIn("required", markdown)
```